nexus/src/app/iam.rs

```diff
@@ -63,15 +63,34 @@ impl super::Nexus {
         &self,
         opctx: &OpContext,
         pagparams: &DataPageParams<'_, Uuid>,
+        group_id: &Option<Uuid>,
     ) -> ListResultVec<db::model::SiloUser> {
         let authz_silo = opctx
             .authn
             .silo_required()
             .internal_context("listing current silo's users")?;
         let authz_silo_user_list = authz::SiloUserList::new(authz_silo.clone());
-        self.db_datastore
-            .silo_users_list_by_id(opctx, &authz_silo_user_list, pagparams)
-            .await
+
+        if let &Some(gid) = group_id {
+            let (.., authz_group, _db_group) =
+                LookupPath::new(opctx, &self.db_datastore)
+                    .silo_group_id(gid)
+                    .fetch()
+                    .await?;
+
+            self.db_datastore
+                .silo_group_users_list(
+                    opctx,
+                    &authz_silo_user_list,
+                    pagparams,
+                    &authz_group,
+                )
+                .await
+        } else {
+            self.db_datastore
+                .silo_users_list(opctx, &authz_silo_user_list, pagparams)
+                .await
+        }
```

Comment:
I'm splitting the logic here, but really these could be two Nexus functions. In fact, I'll make that change. I'm remembering the wisdom of Sandi Metz: move logic forks as high in the call stack as you can.

Reply:
Not sure about the quote but it does make sense to me to break this down.